pkg/server/createExperiment.go

```diff
@@ -1,3 +1,130 @@
+
+type CreateExperiment struct {
+	Version                string                 `json:"version"`
+	ExperimentName         string                 `json:"experiment_name"`
+	ClusterName            string                 `json:"cluster_name"`
+	PerformanceProfile     string                 `json:"performance_profile"`
+	Mode                   string                 `json:"mode"`
+	TargetCluster          string                 `json:"target_cluster"`
+	KubernetesObjects      []KubernetesObject     `json:"kubernetes_objects"`
+	TrialSettings          TrialSettings          `json:"trial_settings"`
+	RecommendationSettings RecommendationSettings `json:"recommendation_settings"`
+}
+
+type KubernetesObject struct {
+	Type       string      `json:"type"`
+	Name       string      `json:"name"`
+	Namespace  string      `json:"namespace"`
+	Containers []Container `json:"containers"`
+}
+
+type Container struct {
+	ContainerImage string `json:"container_image_name"`
+	ContainerName  string `json:"container_name"`
+}
+
+type TrialSettings struct {
+	MeasurementDuration string `json:"measurement_duration"`
+}
+
+type RecommendationSettings struct {
+	Threshold string `json:"threshold"`
+}
+
+func LoadValues(requestName string, deployments map[string][]string, context context.Context) {
+
+	var reqBody CreateExperiment
+	var kubeObj KubernetesObject
+	var containerDataClean []string
+	var requestBody CreateExperiment
+
+	containerMap := make(map[string][]Container)
+
+	//get containers
+	for deployment, containerData := range deployments {
+		containerDataClean = helpers.RemoveDuplicate(containerData)
+		for _, contData := range containerDataClean {
+			containerMap[deployment] = append(containerMap[deployment], Container{ContainerName: contData})
+		}
+	}
+
+	for deployment, containers := range containerMap {
+		for _, con := range containers {
+			singleContainer := []Container{con}
+
+			//parse deployment data
+			requestBody = CreateExperiment{
+				Version:            "1.0",
+				ExperimentName:     fmt.Sprintf("%s-%s-%d", requestName, deployment, con.ContainerName),
+				ClusterName:        reqBody.ClusterName,
+				PerformanceProfile: "resource-optimization-acm",
+				Mode:               "monitor",
+				TargetCluster:      "remote",
+				KubernetesObjects: []KubernetesObject{
+					{
+						Type:       "deployment",
+						Name:       deployment,
+						Namespace:  kubeObj.Namespace,
+						Containers: singleContainer,
+					},
+				},
+				TrialSettings: TrialSettings{
+					MeasurementDuration: "60min",
+				},
+				RecommendationSettings: RecommendationSettings{
+					Threshold: "0.1",
+				},
+			}
+
+			requestBodies := []CreateExperiment{requestBody}
+			err := createExperiment(requestBodies, context)
+			for err != nil {
+				timeToSleep := 30 * time.Second
+				klog.Errorf("Cannot create createExperiment %s in kruize: %v. Will retry in %s\n", requestBodies[0].ExperimentName, err, timeToSleep)
+				time.Sleep(timeToSleep)
```

Comment:
Won't it get stuck here if it fails once?

Reply:
It is not retrying the createExperiment call.

Reply:
Added retry mechanism.